### PR TITLE
refactor: Big refactor

### PR DESCRIPTION
**Pull request by odscjames**
See Changelog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,14 +16,10 @@ pip install -e .
 
 ### Running the command line tool
 
-Call `libcovebods` and pass the filename of some JSON data.
+Call `libcovebods`.
 
-    libcovebods tests/fixtures/0.1/basic_1.json
+    libcovebods -h tests/fixtures/0.1/basic_1.json
```
Comment (Bjwebb) [inline README.md:21]: I find `-h` with a filename kind of odd here.

Comment (odscjames) [reply README.md:21]: Remove tests/fixtures/0.1/basic_1.json and leave -h

**Review by Bjwebb — CHANGES_REQUESTED** [review]

File: libcovebods/jsonschemavalidate.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                "%s is not valid under any of the given schemas"
+                % (json.dumps(instance, sort_keys=True, default=decimal_default),),
+                context=all_errors,
+            )
+
+    more_valid = [s for i, s in subschemas if validator.is_valid(instance, s)]
+    if more_valid:
+        more_valid.append(first_valid)
+        reprs = ", ".join(repr(schema) for schema in more_valid)
+        yield ValidationError("%r is valid under each of %s" % (instance, reprs))
```
[the remark this answers is not in the document]
Comment (odscjames) [reply libcovebods/jsonschemavalidate.py:93]: Yes

**Review by Bjwebb — APPROVED** [review]

**Comment by Bjwebb** [comment]
I've approved, but we need to make sure others are aware of the commandline changes.

**Pull request merged**